src/use-swr.ts

```diff
@@ -393,7 +393,7 @@ function useSWR<Data = any, Error = any>(
     const softRevalidate = () => revalidate({ dedupe: true })
 
     // trigger a revalidation
-    if (!config.initialData) {
+    if (!config.initialData && config.revalidateOnMount) {
```

Comment:
I think this should be an `||` so that if revalidateOnMount is set to true, it always revalidates regardless if there is initialData.
In that sense perhaps the default value of revalidateOnMount should be undefined, so that:
if value is undefined (ie not explicitly set) -> preserve existing behavior
if value is true -> always revalidate
if value is false -> don't revalidate

thoughts?

Reply:
Hi there,
if I understood well, current behavior is that whenever `initialData` is not set revalidation occurs (so that's why `true` is default value).

So if I change that condition with an `||`, I don't set `initialData` and I set `revalidateOnMount` to `false` that condition would be equal to:
`if (!undefined || false)` which is equal to `if (true || false)` and so it would trigger revalidation, ignoring the fact that `revalidateOnMount` flag is set to false.

Reply:
yeah, I didn't explain myself well enough, if you agree on the behavior for the value (undefined, true, false) then the solution isn't the || :)
Probably something like:
`if(config.revalidateOnMount === true || (config.revalidateOnMount === undefined && !config.initialData))`

Reply:
Ah gotcha, well...do you think is needed another value?
What's the downside of setting `true` as default value?

Anyway, yes, it's feasible.

Reply:
I just think that the variable name might be confusing, since it won't revalidate if initialData is provided, so from api usage perspective if I'm setting it as true explicitly (even with initialData), and it's not happening, it's weird, hence my suggestion to have it as undefined by default = current behavior.

Reply:
Yeah you're right. I got your point!
I just pushed the changes with updated code, readme and tests.
Let me know what you think about it!